api/lib/source/cosmosV0-source.js

```diff
@@ -224,14 +224,106 @@ class CosmosV0API extends RESTDataSource {
       validator.signing_info = signingInfos[consensusAddress]
     })
 
-    return validators.map((validator) =>
+    const validatorsWithoutProfiles = validators.map((validator) =>
       this.reducers.validatorReducer(
         this.network.id,
         signedBlocksWindow,
         validator,
         annualProvision
       )
     )
+    if (profile) {
+      return { validators, validatorsWithoutProfiles }
+    } else {
+      return validatorsWithoutProfiles
+    }
+  }
+
+  async getProfilesForValidators(
+    validators,
+    validatorsWithoutProfiles,
+    fiatCurrency = 'USD'
+  ) {
+    validatorsWithoutProfiles = this.getRanksForValidators(
+      validatorsWithoutProfiles
+    )
+    return await Promise.all(
+      validatorsWithoutProfiles.map(async (enrichedValidator) => {
+        const [validatorProfile, allValidatorDelegations] = await Promise.all([
+          this.db.getValidatorProfile(enrichedValidator.operatorAddress),
+          this.getAllValidatorDelegations(enrichedValidator)
+        ])
+        const validator = validators.find(
+          ({ operator_address }) =>
+            operator_address === enrichedValidator.operatorAddress
+        )
+        const fiatValuesResponse = await this.fiatValuesAPI.calculateFiatValues(
+          [
+            {
+              amount: enrichedValidator.tokens,
+              denom: this.network.stakingDenom
+            }
+          ],
+          fiatCurrency
+        )
+        const totalStakedAssets = fiatValuesResponse[this.network.stakingDenom]
+        return this.reducers.validatorProfileReducer(
+          enrichedValidator,
+          validator,
+          validatorProfile,
+          totalStakedAssets,
+          allValidatorDelegations.length,
+          this.network
+        )
+      })
+    )
+  }
+
+  getRanksForValidators(validators) {
+    return validators
+      .sort((a, b) => {
+        const A = new BigNumber(a.tokens)
+        const B = new BigNumber(b.tokens)
+        return A.lt(B) ? 1 : -1
+      })
+      .map((validator, index) => ({
+        ...validator,
+        rank: ++index
+      }))
+  }
+
+  async getValidatorProfile(operatorAddress) {
+    if (!this.validatorsWithProfiles) {
+      const {
+        validators,
+        validatorsWithoutProfiles
+      } = await this.getAllValidators(this.blockHeight, true)
+      const validatorsWithProfiles = await this.getProfilesForValidators(
+        validators,
+        validatorsWithoutProfiles
+      )
+      this.validatorsWithProfiles = _.keyBy(
```

Comment:
why is this a different object then the validators and why is it not stored in the store?

Reply:
it gets stored later right? Let me check the code. Should be